File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include<string.h>
#include<ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
string text= get_string("Text: ");
float l = count_letters(text);
float m = count_words(text)+1.0;
float n = count_sentences(text);
float L = (l/m) * 100.0;
float S = (n/m) * 100;
int grade = round(0.0588*L - 0.296*S - 15.8);
if (1<= grade && grade<=16)
{
printf("Grade %i\n", grade);
}
else if (grade<1)
{
  printf("Before Grade 1\n");
}
else
{
  printf("Grade 16+\n");
}

}

int count_letters(string text)
{
int numberofalphabets=0;
for (int x=0; x<strlen(text); x++)
{
  if(isalpha(text[x]))
  {
    numberofalphabets++;
  }
}
return numberofalphabets;
}

int count_words(string text)
{
int numberofwords=0;
for (int x=0; x<strlen(text); x++)
{
  if(isspace(text[x]))
  {
    numberofwords++;
  }
}
return numberofwords;
}

int count_sentences(string text)
{
  int numberofwords=0;
for (int x=0; x<strlen(text); x++)
{
  if(text[x]== 46)
  {
    numberofwords++;
  }
    if(text[x]== 33)
  {
    numberofwords++;
  }
  if(text[x]== 63)
  {
    numberofwords++;
  }
}
return numberofwords;
}

File: sentimental-readability/readability.py
from cs50 import get_string
def count_letters(text):
    noofalpha=0
    for x in text:
        if(x.isalpha() == True):
            noofalpha+=1
    return int(noofalpha)
def count_words(text):
    noofwords = 0
    for x in text:
        if (x.isspace()) == True:
            noofwords+=1
    noofwords+=1
    return int(noofwords)
def count_sentences(text):
    noofsent=0
    for x in text:
        if (x.find(chr(63)) != -1):
            noofsent+=1
        if (x.find(chr(46)) != -1):
            noofsent+=1
        if (x.find(chr(33)) != -1):
            noofsent+=1
    return int(noofsent)


text = get_string('Text: ')
l = count_letters(text)
m = count_words(text)
n = count_sentences(text)
L = (l/m)*100
S = (n/m)*100
grade = round(0.0588*L - 0.296*S - 15.8)
if (1<= grade and grade<=16):
    print(f"Grade {grade}")
elif(grade<1):
    print("Before Grade 1")
else:
    print("Grade 16+")
